drivers/spi/spi.go

```diff
@@ -0,0 +1,44 @@
+package spi
+
+import (
+	"golang.org/x/exp/io/spi"
+	"sync"
+	"golang.org/x/exp/io/spi/driver"
+	"fmt"
+)
+
+type Connection struct {
+	connection *spi.Device
+	address int
+	mutex *sync.Mutex
+}
+
+// NewConnection creates and returns a new connection to a specific
+// spi device on a bus and address
+func NewConnection(opener driver.Opener, address int) (connection *Connection) {
+	dev, err := spi.Open(opener)
+	if err != nil {
+		panic(err)
+	}
+
+	return &Connection{connection: dev, address: address, mutex: &sync.Mutex{}}
+}
+
+// TODO: Read - the exp/io/spi implementation doesn't currently support reading
```

Comment:
exp/io/spi does support reading, I've got a little of that worked out in my fork. The second parameter in TX() should be the same size as the first parameter and it will hold the results of request, if there is one.